xonsh/tools.py

```diff
@@ -583,7 +572,7 @@ def ensure_int_or_slice(x):
     """Makes sure that x is list-indexable."""
     if x is None:
         return slice(None)
-    elif is_int(x):
+    elif isinstance(x, int):
```

Comment:
about what exactly? i'm currently working on a patch that allows more versatile to define ensurer functions and constraints.